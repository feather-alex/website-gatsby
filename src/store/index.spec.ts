import { createMemoryHistory, History } from 'history';
import { createStore, Reducer } from 'redux';
import createRootReducer from './reducer';

// mock out calls to UUID library coming from cart.reducer.ts initialState
jest.mock('uuid', () => ({
  v1: () => 'mock-uuid-string-for-testing'
}));

describe('Redux Store', () => {
  let memoryHistory: History;
  let reducer: Reducer;

  beforeAll(() => {
    // use memory history, as this test is not running in a real browser
    memoryHistory = createMemoryHistory({ keyLength: 0 });
  });

  beforeEach(() => {
    reducer = createRootReducer(memoryHistory);
  });

  it('should be able to create the store', () => {
    expect(createStore(reducer)).toBeDefined();
  });

  it('should ensure there has not been any schema changes to the redux store', () => {
    const store = createStore(reducer);
    // ensure that the redux state schema has not changed
    // by comparing the initial state produced against our snapshot
    expect(store.getState()).toMatchSnapshot();
  });
});
