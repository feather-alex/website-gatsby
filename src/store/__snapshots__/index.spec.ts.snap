// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux Store should ensure there has not been any schema changes to the redux store 1`] = `
Object {
  "about": Object {
    "allDepartments": null,
    "error": false,
    "isFetching": false,
  },
  "accounts": Object {
    "accountHistory": Object {
      "charges": Array [],
      "error": null,
      "hasMore": false,
      "isFetching": false,
      "perPage": 7,
    },
    "accountOverview": Object {
      "error": null,
      "isFetching": false,
      "overview": Object {
        "currentPlan": Object {
          "membershipFee": 0,
          "startDate": "",
          "type": "Legacy",
        },
        "deliveryAreaIdentifier": null,
        "discounts": Array [],
        "endDate": "",
        "items": Array [],
        "leases": Array [],
        "monthlySubtotal": 0,
        "numberOfFreeTrips": 0,
        "requestedDeliveryDate": null,
        "startDate": "",
        "subscriptionId": 0,
        "taxRate": 0,
      },
    },
    "billingInformation": Object {
      "defaultSource": Object {
        "expMonth": 0,
        "expYear": 0,
        "id": "",
        "lastFour": "0",
        "sourceType": "",
      },
      "error": null,
      "isFetching": false,
      "sources": Array [],
      "startDate": null,
    },
    "changePassword": Object {
      "error": null,
      "isFetching": false,
      "isPasswordUpdated": false,
    },
    "personalInformation": Object {
      "accountLastAccessedAt": "",
      "addressInfo": Array [],
      "email": "",
      "error": null,
      "firstName": "",
      "isFetching": false,
      "lastName": "",
      "phone": "",
    },
  },
  "app": Object {
    "contentfulPages": Object {
      "error": null,
      "isFetching": false,
      "pages": Array [],
    },
    "navbar": Object {
      "bannerColor": "",
      "bannerMessage": "",
      "bannerType": null,
      "content": null,
      "isBannerVisible": false,
      "isContentLoading": false,
    },
    "newsletter": Object {
      "displaySuccess": false,
      "email": null,
      "error": null,
      "isFetching": false,
    },
    "overlay": Object {
      "is3DOverlayOpen": false,
      "isAddItemsToCurrentPlanOverlayOpen": false,
      "isDeliveryOverlayOpen": false,
      "isDimensionsOverlayOpen": false,
      "isEditQuizResultsOverlayOpen": false,
      "isFailedSSNOverlayOpen": false,
      "isMembershipOverlayOpen": false,
      "isMembershipSelectedOverlayOpen": false,
      "isMiniCartOpen": false,
      "isMobileNavOpen": false,
      "isMobileProductFiltersOpen": false,
      "isNoSSNOverlayOpen": false,
      "isPlanInformationOverlayOpen": false,
      "isPlanSelectionOverlayOpen": false,
      "isProductInfoOverlayOpen": false,
      "isQuizOverlayOpen": false,
    },
    "productEntities": Object {
      "data": null,
      "error": null,
      "isFetching": false,
    },
    "windowDimensions": Object {
      "height": 0,
      "width": 0,
    },
  },
  "auth": Object {
    "forgotPassword": Object {
      "email": "",
      "error": null,
      "hasSentResetPasswordLink": false,
      "isFetching": false,
    },
    "login": Object {
      "authenticated": false,
      "email": "",
      "error": null,
      "hasViewedWelcomeModal": false,
      "isFetching": false,
      "isFirstLogin": false,
      "name": "",
    },
    "register": Object {
      "email": "",
      "error": null,
      "hasRegistered": false,
      "isFetching": false,
    },
    "verification": Object {
      "error": null,
      "hasResentVerification": false,
      "isFetching": false,
    },
  },
  "cart": Object {
    "error": null,
    "isFetching": false,
    "isRecommendationsFetching": false,
    "items": Array [],
    "promo": null,
    "promoError": null,
    "promoState": "empty",
    "recommendations": Array [],
    "recommendationsError": null,
    "unavailableItems": Array [],
    "uuid": "mock-uuid-string-for-testing",
  },
  "checkout": Object {
    "amountError": null,
    "billingAddressInfo": Object {
      "billingApt": "",
      "billingCity": "",
      "billingPostalCode": "",
      "billingState": "",
      "billingStreetAddress": "",
      "googleBillingStreetAddress": "",
    },
    "cardError": Object {
      "error": null,
    },
    "customerInfo": Object {
      "company": "",
      "email": "",
      "firstName": "",
      "lastName": "",
      "persona": null,
    },
    "deliveryFee": 0,
    "deliveryInfo": Object {
      "apt": "",
      "city": "",
      "googleDeliveryStreetAddress": "",
      "phone": "",
      "state": "",
      "streetAddress": "",
      "zipcode": "",
    },
    "depositAmount": 0,
    "depositError": null,
    "dueNow": 0,
    "generateNewStripeToken": false,
    "isCreditNotFound": false,
    "isDeliverySameAsBilling": true,
    "isFetchingAmount": false,
    "isPlacingOrder": false,
    "isSSNNotFound": false,
    "isSSNNotValid": false,
    "isSubmittingDeposit": false,
    "itemsError": null,
    "maxTCVError": null,
    "membershipFee": 0,
    "monthlyTax": 0,
    "monthlyTotal": 0,
    "orderTCV": 0,
    "serverError": null,
    "ssnInfo": Object {
      "legalFirstName": "",
      "legalLastName": "",
      "ssn": "",
    },
    "step": "customer-info",
    "taxDueNow": 0,
  },
  "contact": Object {
    "displayErrorMessage": false,
    "displaySuccessMessage": false,
    "isProccessingRequest": false,
  },
  "enterprise": Object {
    "error": null,
    "faqs": Array [],
    "heroLockup": null,
    "horizontalLockup": null,
    "horizontalLockup2": null,
    "isFetching": false,
    "meta": null,
    "productShowcase": null,
    "titleButtonLockup": null,
    "titledTripleVerticalImageLockup": null,
  },
  "entities": Object {
    "packages": Object {
      "data": Array [],
      "error": null,
      "isFetching": true,
    },
    "pkg": Object {
      "data": Object {
        "availability": Array [
          Object {
            "deliveryArea": null,
            "isEnabled": false,
            "isInStock": false,
            "stockExpectedDate": null,
          },
        ],
        "category": Object {
          "identifier": "",
          "name": "",
        },
        "description": "",
        "identifier": "",
        "lifestyle": Object {
          "image": Object {
            "desktop": null,
            "mobile": null,
          },
          "summary": "",
        },
        "listingImage": Object {
          "desktop": null,
          "mobile": null,
        },
        "options": Array [],
        "otherImages": Array [],
        "title": "",
        "variants": Array [],
      },
      "error": null,
      "isFetching": true,
    },
  },
  "faq": Object {
    "error": null,
    "faqCategories": Array [],
    "isFetching": false,
    "meta": Object {
      "description": "",
      "imageUrl": "",
      "name": "",
      "title": "",
    },
  },
  "featherPerks": Object {
    "error": null,
    "isFetching": false,
    "perks": null,
  },
  "form": Object {},
  "homepage": Object {
    "bestSellers": null,
    "error": null,
    "hero": null,
    "isFetching": false,
    "meta": null,
    "reviews": null,
    "sections": null,
    "shopByRoom": null,
    "textLockup": null,
  },
  "howItWorks": Object {
    "error": null,
    "faqs": Array [],
    "header": "",
    "isFetching": false,
    "meta": Object {
      "description": "",
      "imageUrl": "",
      "name": "",
      "title": "",
    },
    "steps": Array [],
  },
  "locationContent": Object {
    "ctaSection": null,
    "error": null,
    "hero": null,
    "horizontalLockup": null,
    "isFetching": false,
    "meta": null,
    "pointsBreakdown": null,
  },
  "navigation": Object {
    "customerCanCheckout": false,
    "deliverToPostal": false,
    "deliveryAreaIdentifier": null,
    "displayEmailInput": false,
    "displayInitialGreeting": true,
    "displayNavigationBanner": false,
    "displayNoDelivery": false,
    "displayPostalInput": false,
    "displayWeDeliver": false,
    "hasEnteredEmail": false,
    "invalidPostalInput": false,
    "postal": null,
  },
  "plan": Object {
    "cartMinimum": null,
    "deliveryAreaIdentifier": null,
    "deliveryFee": null,
    "deliveryZipcode": null,
    "error": null,
    "isEmailSubmitted": false,
    "isFetching": false,
    "isInDeliveryZone": null,
    "membershipState": "NONE",
    "monthlyMembershipFee": null,
    "rentalLength": null,
  },
  "productDetails": Object {
    "data": Object {
      "3dAssetId": null,
      "availability": Array [],
      "brand": Object {
        "identifier": "",
        "image": Object {
          "desktop": null,
          "mobile": null,
        },
        "name": "",
      },
      "categories": Array [],
      "description": "",
      "identifier": "",
      "lifestyle": Object {
        "image": Object {
          "desktop": null,
          "mobile": null,
        },
        "summary": "",
      },
      "materials": Array [],
      "options": Array [],
      "styles": Array [],
      "subclass": Object {
        "identifier": "",
        "name": "",
      },
      "title": "",
      "variants": Array [],
    },
    "error": null,
    "isFetching": false,
  },
  "productList": Object {
    "error": null,
    "isFetching": false,
    "meta": Object {
      "availableFilters": Object {
        "brands": Array [],
        "classes": Array [],
        "subclasses": Array [],
      },
      "pageNumber": 0,
      "total": 0,
    },
    "products": Array [],
  },
  "productPairings": Object {
    "bestsellers": Array [],
    "error": null,
    "isFetching": false,
    "products": Array [],
  },
  "quiz": Object {
    "budget": "none",
    "currentBedroomIndex": 0,
    "currentStepIndex": 0,
    "dateChanged": null,
    "deliveryArea": "all",
    "direction": "forward",
    "email": "",
    "isChanged": false,
    "isSubmitted": false,
    "locationPathBeforeQuizOpened": "/",
    "name": "",
    "numberOfBedrooms": 0,
    "reason": "none",
    "rooms": Object {
      "bedrooms": Array [
        Object {
          "bedSize": "",
        },
      ],
      "diningRoom": Object {
        "chairs": "",
      },
      "homeOffice": Object {
        "workFrequency": "",
      },
      "livingRoom": Object {
        "function": "",
        "roomSize": "",
      },
    },
    "selectedImages": Array [],
    "selectedRooms": Array [],
    "steps": Array [
      "select-location",
      "why",
      "what-rooms",
    ],
    "styles": Array [],
    "uuid": "",
  },
  "quizPackages": Object {
    "activeQuizRoom": null,
    "data": null,
    "error": null,
    "isFetching": false,
  },
  "router": Object {
    "action": "POP",
    "location": Object {
      "hash": "",
      "pathname": "/",
      "query": Object {},
      "search": "",
      "state": undefined,
    },
  },
  "search": Object {
    "keyword": "",
    "packages": Object {
      "data": Array [],
      "error": null,
      "isFetching": false,
    },
    "products": Object {
      "data": Array [],
      "error": null,
      "isFetching": false,
      "isInfiniteLoading": false,
      "offset": 0,
      "total": 0,
    },
  },
  "trackingParameters": Object {},
}
`;
